# Vorgehen

1. Modell (z.B. $X_i i.i.d \sim \mathcal{N}(\mu, \sigma^2)$
2. Nullhypothese (z.B. \mu=\mu_0=70)
3. Verteilung der Teststatistik unter $H_0$ (z.B. $\overline{X}_{12} \sim \mathcal{N}(70, \frac{1.5^2}{12}$)
4. Signifikanzniveau (z.B. $\alpha=0.05=5\%$)
5. Verwerfungsbereich für die Teststatistik (z.B. mittels `norm.ppf`)
6. Testentscheid

# 6.1

Nullhypothese $ H_0: X \sim \mathcal{N}(70, 1.5^2), \mu = 70, \sigma=1.5 $ (in Centiliter)

Alternativhypothese $ H_A: \mu < 70 $ (zu _wenig_ abgefüllt)

Signifikanzniveau: $ 5\% $

In [3]:
import math
import numpy as np
from scipy.stats import norm

significance = 0.05
pop_mean = 70
pop_std = 1.5
sample = np.array([71, 69, 67, 68, 73, 72, 71, 71, 68, 72, 69, 72])

upper = norm.ppf(q=significance, loc=pop_mean, scale=pop_std/math.sqrt(len(sample)))
print('Verwerfungsbereich: (-∞,{:.2f}]'.format(upper))

sample_mean = sample.mean()
sample_std = sample.std(ddof=1)
print('sample mean={:.3f}, sample std={:.3f}'.format(sample_mean, sample_std))

# the population standard deviation is known: z-test
p = norm.cdf(x=sample_mean, loc=pop_mean, scale=pop_std / math.sqrt(len(sample)))
print('P(X<=70)={:.3f}'.format(p))

Verwerfungsbereich: (-∞,69.29]
sample mean=70.250, sample std=1.960
P(X<=70)=0.718


Die Nullhypothese trifft mit einer Wahrscheinlichkei von $71.8\%$ zu und kann darum bei einem Signifikanzniveau von $5\%$ nicht verworfen werden. Man kann nicht behaupten, dass der Weinhändler zu wenig abfüllt. (Die Stichprobe liesse eher vermuten, dass der Weinhändler eher zu viel abfüllt.)

# 6.2

Nullhypothese $H_0: X \sim \mathcal{N}(70, \sigma^2), \mu=70, \sigma$ unbekannt.

Alternativhypothese $H_A: \mu < 70$ (zu wenig abgefüllt)

Signifikanzniveau: $5\%$

In [4]:
import math
import numpy as np
from scipy.stats import t

pop_mean = 70
sample = np.array([71, 69, 67, 68, 73, 72, 71, 71, 68, 72, 69, 72])
n = len(sample)

upper = t.ppf(q=significance, df=n-1, loc=pop_mean, scale=sample_std/math.sqrt(n))
print('Verwerfungsbereich: (-∞,{:.2f}]'.format(upper))

sample_mean = sample.mean()
sample_std = sample.std(ddof=1)
print('sample mean={:.3f}, sample std={:.3f}'.format(sample_mean, sample_std))

p = t.cdf(x=sample_mean, loc=pop_mean, df=len(sample)-1, scale=sample_std / math.sqrt(len(sample)))
print('P(X<=70)={:.3f}'.format(p))

# standardized
p = t.cdf(x=(sample_mean-pop_mean)/(sample_std/math.sqrt(len(sample))), df=len(sample)-1)
print('P(X<=70)={:.3f}'.format(p))

Verwerfungsbereich: (-∞,68.98]
sample mean=70.250, sample std=1.960
P(X<=70)=0.666
P(X<=70)=0.666


Die Nullhypothese trifft mit einer Wahrscheinlichkeit von $66.6\%$ zu und kann darum bei einem Signifikanzniveau von $5\%$ nicht verworfen werden. Man kann nicht behaupten, dass der Weinhändler zu wenig abfüllt.

Gegenüber dem z-Test in Aufgabe 6.1 war hier die Standardabweichung nicht bekannt und musste geschätzt werden. Es wurde somit ein t-Test durchgeführt.

# 6.3

$ n = 16, X_i $ in $ \mu g, \overline{x}_{16} = 204.2 \mu g$, Grenzwert $= 200 \mu g$, Signifikanzniveau $=0.05$

## a)

Annahme: $ \sigma=10\mu g $

Nullhypothese $ H_0: X \sim \mathcal{N}(200, 10^2) $

Alternativhypothese $ H_A: \mu > 200 $ (Grenzwert überschritten)

In [8]:
import math
from scipy.stats import norm

n = 16
pop_mean = 200
pop_std = 10
sample_mean = 204.2

lower = norm.ppf(q=0.95, loc=pop_mean, scale=pop_std/math.sqrt(n))
print('Verwerfungsbereich: [{:.2f},∞)'.format(lower))

p = norm.cdf(x=sample_mean, loc=pop_mean, scale=pop_std/math.sqrt(n))
print('P(X>200)={:.3f}'.format(p))

Verwerfungsbereich: [204.11,∞)
P(X>200)=0.954


Eine Grenzwertüberschreitung trifft mit $95.5\%$ Wahrscheinlichkeit zu. Bei einem Signifikanzniveau von $5\%$ kann die Nullhypothese damit verworfen werden. Es liegt wahrscheinlich eine Grenzwertüberschreitung vor.

## b)

$\mu=205 \mu g$

In [9]:
import math
from scipy.stats import norm

real_mean = 205
fake_mean = 200
n = 16
pop_std = 10

# real solution: x from exercise a)
p = 1 - norm.cdf(x=204.11, loc=205, scale=pop_std/math.sqrt(n))
print('P={:.3f}'.format(p))

# simulation
runs = 5000
rejected = 0
for i in range(0, runs):
    sample = norm.rvs(size=n, loc=real_mean, scale=pop_std)
    p = norm.cdf(x=sample.mean(), loc=fake_mean, scale=pop_std/math.sqrt(n))
    if p > 0.95:
        rejected += 1

p = rejected / runs
print(p)

P=0.639
0.6314


Eine Simulation mit 5000 Durchläufen ergab, dass die Grenzwertüberschreitung in ca. $64.4\%$ der Fälle entdeckt werden konnte.

## c)

$\mu = 200\mu g$

In [12]:
import math
from scipy.stats import norm

real_mean = 200
n = 16
pop_std = 10

# simulation
runs = 50000
rejected = 0
for i in range(0, runs):
    sample = norm.rvs(size=n, loc=real_mean, scale=pop_std)
    p = norm.cdf(x=sample.mean(), loc=real_mean, scale=pop_std/math.sqrt(n))
    if p > 0.95:
        rejected += 1

p = rejected / runs
print(p)

0.05134


Per Definition: $5\%=0.05$ (Niveau des Tests). Die Simulation nähert sich diesem Wert mit steigender Anzahl Durchläufen an. (Wahrscheinlichkeit für einen _false positive_.)

## d)

In [14]:
import math
from scipy.stats import t

pop_mean = 200
n = 16
sample_mean = 204.2
sample_std = 10

lower = t.ppf(q=0.95, df=n-1, loc=pop_mean, scale=sample_std/math.sqrt(n))
print(lower)

p = t.cdf(x=sample_mean, loc=pop_mean, df=n-1, scale=sample_std/math.sqrt(n))
print(p)

204.38262588923138
0.9431708331031162


Eine Grenzwertüberschreitung liegt mit einer Wahrscheinlichkeit von $94.3\%$ vor. Bei einem Signifikanzniveau von $5\%$ kann die Nullhypothese deshalb _nicht verworfen_ werden.

## e)

Die Stichprobe könnte zu klein sein. Je grösser die Stichprobe, desto stärker der t-Test.